### DISCUSSION TITLE
End G-Code for Custom Bowden Extruder

### DISCUSSION THREAD

**Discussion by speendo** [supermerill/SuperSlicer#2508] [Q&A]
Coming from PrusaSlicer with a strongly customized configuration for my P3Steel with Bowden Extruder, I am using SuperSlicer for a few days.

I am very astonished with the print quality that I could achieve by just using the default configuration.

However, when the print is finished, I ended up with a clogged nozzle almost every time. First I thought this was just an accident, then I thought there might be a problem in my hardware.

Finally, I looked at the End G-Code (which I didn't touch) and now I got a grasp on what's going wrong. This is the default End G-Code

```
G1 E-4 F2100.00000
G91
G1 Z1 F7200.000
G90
G1 X195 Y1
G1 X190 E4
G1 F4000
G1 X180 E2.7  
G1 F4600
G1 X110 E2.8
G1 F5200
G1 X40 E3  
G1 E-15.0000 F5000
G1 E-50.0000 F5400
G1 E-15.0000 F3000
G1 E-12.0000 F2000
G1 F1600
G1 X0 Y1 E3.0000
G1 X50 Y1 E-5.0000
G1 F2000
G1 X0 Y1 E5.0000
G1 X50 Y1 E-5.0000
G1 F2400
G1 X0 Y1 E5.0000
G1 X50 Y1 E-5.0000
G1 F2400
G1 X0 Y1 E5.0000
G1 X50 Y1 E-3.0000
G4 S0
M107 ; turn off fan
{if layer_z < max_print_height}G1 Z{z_offset+min(layer_z+30, max_print_height)}{endif} ; Move print head up
M104 S0 ; turn off temperature
M140 S0 ; turn off heatbed
G28 X0  ; home X axis
M84     ; disable motors
```
These wild retractions and extrusions might work on a good direct drive (but I am not even sure about that) but they are dedicated to clogg a bowden extruder.

Moreover I don't even understand their purpose. I used to only retract the filament a little bit and that's it.

Is there a deeper meaning in these moves?

**Comment by supermerill** [comment]
> Is there a deeper meaning in these moves?

It's to shape the filament tip, to avoid stringing inside the filament path.
Isn't it the same with prusaslicer?

**Reply by speendo** [reply]
I just wanted to write that it might be the same in PrusaSlicer. But then I looked it up and it seems to be different.

This is from my PrusaSlicer for a generic printer
```
M104 S0 ; turn off temperature
G28 X0  ; home X axis
M84     ; disable motors
```

**Reply by supermerill** [reply]
here is what i get by creating a new generic printer (as end gcode):
```
M104 S0 ; turn off temperature
G28 X0  ; home X axis
M84     ; disable motors
```
so it's the same as the prusaslicer, for what I can see.

How did you manage to get the mmu end gcode  into your end gcode?

**Reply by speendo** [reply]
That's a really good question. I cannot reproduce it.
